api/app/files/api.py

```diff
@@ -45,9 +46,33 @@ def get(self):
             return send_file(temp.name, as_attachment=True, attachment_filename=args['filename'], mimetype='application/pdf')
 
 
+
+class FileAPIImproved(FileUploadMixin, restful.Resource):
+    def get(self, file_id):
+        # TO DO: Query the File table
+
+        req_parser = reqparse.RequestParser()
+        req_parser.add_argument('filename', type=str, required=True)
+        args = req_parser.parse_args()
+        LOGGER.debug("FileUpload GET args: {}".format(args))
+
+        bucket = _get_storage_bucket()
+
+        blob = bucket.blob(args['filename'])
+
+        # TO DO : Rename the file and set the appropriate mime_type
+
+        with tempfile.NamedTemporaryFile() as temp:
+            blob.download_to_filename(temp.name)
+            return send_file(temp.name, as_attachment=True, attachment_filename=args['filename'], mimetype='application/pdf')
+
+
     @auth_required
     def post(self):
         args = self.req_parser.parse_args()
+        file_name = args['filename']
+        mime_ = args['event_id']
+        email_template = args['email_template']

```

Comment:
So I think look at the existing `FileUploadAPI` `post` method and do something similar. You can probably then get the file name by doing `file_name = file.filename` and them mime_type by doing `mime_type = file.content_type` (this is what's happening currently). Then create your file object with `new_file = File(file_name, mime_type)`. Then again make a new repository method to save this file object. I'll leave this to you to see how we do this else where in the project (if you're still struggling let me know)
Email template just remove.

Reply:
The rest of the saving to blob storage should be the same as what's in the existing post method